lib/bundler/uri_credentials_filter.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+module Bundler
+  module URICredentialsFilter
+    def anonymized_uri(uri_to_anonymize)
+      return uri_to_anonymize if uri_to_anonymize.nil?
+      uri = uri_to_anonymize.dup
+      uri = URI(uri.to_s) unless uri.is_a?(URI)
+      uri.user = uri.password = nil if uri.userinfo
```

Comment:
Okay, that works. Although, I don't think there's a way to differentiate between a user using 'x' or 'x-oauth-basic' as their login/authentication password vs. them using that to specify they want to authenticate using oauth. Though I guess maybe we can just completely hide user and password sections if password is 'x' or 'x-oauth-basic'? This should cover the vast majority of credential screening cases.